sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -717,22 +899,26 @@ class RocksDB(
           numKeys,
           capturedFileMappings,
           columnFamilyMapping,
-          maxColumnFamilyId)) =>
+          maxColumnFamilyId,
+          uniqueId)) =>
         try {
+          versionToUniqueIdLineage = versionToUniqueIdLineage.filter(_._1 >= version)
```

Comment:
Is there a data race to `versionToUniqueIdLineage` between task thread and maintenance thread?

Reply:
This should happen *after* snapshot is uploaded, *and* the one in lineage is not the latest version and matches uniqueID of the one to upload.